src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+
+import datetime
+import pathlib
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Type
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+if TYPE_CHECKING:
+    import torch.utils.tensorboard
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    #: The class that's used to instantiate a summarywriter
+    SummaryWriter: Type['torch.utils.tensorboard.SummaryWriter']
+
+    def __init__(
+        self,
+        experiment_path: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize result tracking via Tensorboard.
+
+        :param experiment_path:
+            The experiment path. A custom path at which the tensorboard logs will be saved.
+        :param experiment_name:
+            The name of the experiment, will be used as a sub directory name for the logging. If no default is given,
+            the current time is used. If set, experiment_path is set, this argument has no effect.
+        :param tags:
+            The additional run details which are presented as tags to be logged
+        """
+        from torch.utils.tensorboard import SummaryWriter as _SummaryWriter
+        self.summary_writer_cls = _SummaryWriter
+        self.tags = tags
+
+        if experiment_path is None:
+            if experiment_name is None:
+                experiment_name = datetime.datetime.now().isoformat()
+            path = PYKEEN_LOGS.joinpath("tensorboard", experiment_name)
+        elif isinstance(experiment_path, str):
+            path = pathlib.Path(experiment_path)
+        self.path = path
```

Comment:
what should happen if the path already exists?

Reply:
It's a very good point and something I am not quite sure how to handle. If the user runs two jobs passing the same directory for tb, you can get some interesting results as seen below:
![Screenshot from 2021-05-06 20-14-16](https://user-images.githubusercontent.com/10208489/117353218-63a93a00-ae9f-11eb-8efd-719a9285a69f.png)
Of course this is desirable behavior if you are resuming training. 

Ideally, the structure should be: `root_exp_dir/exp_1.... root_exp_dir/exp_n.` Then tb could be run at the root dir and allow for comparison between different experiments where some parameters are being changed for example.

Any thoughts on how this should be handled?

Reply:
Where there any thoughts from anyone on how this should be handled? I am inclined towards trusting the user to set the directory correctly but very open to suggestions.

Reply:
I'd rather not over-engineer this. Just have make sure the docs make it clear that it's up to the user to take care of this